### ISSUE TITLE
"ERROR: Invalid key" when trying to share (upload) a profile

### ISSUE THREAD

**Issue by yogensia** [Nexus-Mods/NMM-Test-Builds#2]
For now I can't think of many details to provide, but I haven't been able to upload a profile so far, so the error is consistent on my Win7 x64 machine. A completely clean NMM install and/or running NMM as admin hasn't helped so far.

I see a progress windows with 4 steps, I see it reach the 3rd step, not sure about the 4th because it speeds up a lot there, then the [error shows up](http://i.imgur.com/NpKAESW.png).

**Comment by DuskDweller** [comment]
Let me know if it works now, we had to change a few things remotely.

**Comment by roamwild** [comment]
NMM 0.62.2 https://nexusmodmanager.slack.com/files/paul/F135X3ZHD/nexus_mod_manager-0.62.2.exe
NMM and Games installed outside of Program Files directory
Win7 64 Home, Firefox 45.0.2

**Observed behavior:**
Choose to BackupShare a saved profile named 'roamwild FO4 001' from either Main Toolbar or the Profiles Tab
"Do you want to upload ... you will be taken to a screen where you can ..." click Yes
Progress bar gets to **3/4 Analyzing Request...**
Progress dialog disappears for a couple seconds, followed by **Error: Invalid Key**
Re-trying the 'BackupShare' sometimes shows no action in the application until it's restarted. Same error after restart.
Eventually reached*\* 'too many failed attempts, please wait 15min'**

**Steps taken to work around (failed)**
Cleared browser cache and temp directories, closed browser instances, re-logged into Nexus and NMM.

![invalid key](https://cloud.githubusercontent.com/assets/18669746/14805324/e16a95c0-0b38-11e6-8839-4cecf9ea9f7c.jpg)

**running NMM as Admin**, with different Mod Directories/Profiles under this Windows Account:

Clicked 'BackupShare' from main window Toolbar and chose YES
**Progress got to '4/4 - Completed' but did not show any screen** or bring up a browser window to enter information about the profile.

I found two profiles on my Nexus Activity Feed from these "Completed" Admin attempts but they both seem broken: Clicking either of these items shows me the Nexusmods.com main File page (game list)
http://www.nexusmods.com//profiles/2100/
http://www.nexusmods.com//profiles/2095/
![admintest update](https://cloud.githubusercontent.com/assets/18669746/14805447/f58b305e-0b39-11e6-9433-0d414d85fb8a.jpg)

I then re-started NMM under this Admin account and successfully initiated a Backup and Share on 'admintestshare' from the Profiles Tab. The appropriate page was brought up in my browser.

At this point I was made aware of the proper link/address to find the Profiles on Nexusmods.com, something which was not apparent before.

**Comment by DuskDweller** [comment]
We temporarily removed the 15 minutes cooldown for too many "Invalid key"s, we'd like to find out what's causing this, please let us know your OS (also whether it's 32 or 64 bits), your CPU and you Antivirus/security suite.

At this stage, I think there's nothing you can do to avoid this issue, if you're affected then it's just random whether you'll get an invalid key or not.

**Comment by yogensia** [comment]
The error is still there for me.

I'm on a i7 2600K using Win7 Home Premium 64bit, Chrome 64bit, and Avast Free.

BTW Disabling avast shields doesn't seem to help either.

**Comment by roamwild** [comment]
I get **INVALID KEY most of the time** while attempting to upload or download a Profile.
i.e. attempting to Download Pauls Test 1 https://www.nexusmods.com/fallout4/profiles/1530
Five spammed attempts result in INVALID KEY but the sixth attempt worked.
(Switching to this profile crashes NMM with System.ArgumentNullException: Value (first) cannot be null.)

Windows 7 Home Premium 64-bit SP1
**CPU** Intel Core i7 3770
**Security** Microsoft Security Essentials, Windows Firewall
More detailed information on hardware and OS here http://pastebin.com/u99wYwd1

**Comment by DuskDweller** [comment]
Do you still get invalid key using 0.62.3?

**Comment by yogensia** [comment]
Yeah the issue persists for me, still happening 100% of the time. :confused:

**Comment by DuskDweller** [comment]
Ok, thanks, we'll try something different.

**Comment by DuskDweller** [comment]
Anyone experiencing this issue, can you tell me where is NMM installed ? Is it in the Program Files (x86) folder?
Can you please try this:
run Nexus Mod Manager (get to the main window)
open task manager, check if NexusClient is being run at 32 or 64 bits.

**Comment by yogensia** [comment]
Mine is installed at C:\Program Files\Nexus Mod Manager\, [looks to be running in 64bits](https://dl.dropboxusercontent.com/u/251256/ShareX/PROCEXP64_2016-04-28_15-08-14.png).

Let me know if there's any more info you may need.

**Comment by KunoMochi** [comment]
Just wanted to chime in, this recently happened to me as well on 0.62.5.
**OS:** Windows 7 Ultimate 64-bit SP1
**Security:** Microsoft Security Essentials
**CPU:** AMD Phenom II X4 955

**Comment by DuskDweller** [comment]
Kuno did it just happen once, or you had to retry many times?

**Comment by crowbar01** [comment]
I also got this using different characters such as ( ) < > etc. Repeated. v.0.62.5

**Comment by KunoMochi** [comment]
Dusk, it only happened once and was able to continue after a retry.

Speaking of which, it just happened again when I accidentally tried to restore a profile of mine (Test Profile - Kuno's Landscape Profile) from the server. A second retry went through, though.

**Comment by roamwild** [comment]
C:\Nexus Mod Manager\Nexusclient.exe 0.62.6 **64 bit** process
Still getting this error with great frequency. I used to spam continually but now give a second or two between most re-tries and generally get the error 1 - 4 times before a success. Subsequent retries will continue this behavior.

**Comment by crowbar01** [comment]
This is still happening for me on 0.62.7, it wasn't before, but is now, and nothing has changed except a few new mods added.

**Comment by roamwild** [comment]
0.62.10 
C:\Nexus Mod Manager\Nexusclient.exe 64 bit process
Windows 7 Home Premium 64-bit SP1, MS Security Essentials
Intel Core i7-3770

If anything, this issue seems to be even worse for me these days. Could be just its random/frequent nature though and bad luck. I definitely have noted more continual occurrences (10+ times in a row) lately.

Still getting this error more often than not on any profile share/update/download. About 1 out of 4 operations will succeed at the very best; sometimes I'll get the error 10 - 15 times in a row as I spam attempts, re-trying the same operation. Generally I try 4 times, pause a moment, try a few more times, and usually succeed during these two sets. Sometimes, and rarely, it will only take one or two tries for me to upload/download a profile. Sometimes I'll manage to re-try between progress and error pop-ups and make a double request (cannot reach Nexus Metadata..).

After this Error, on most Update/Upload or Restore/Download attempts, the automatic 'Refreshing Online Profiles List' triggers. Maybe it's better to always have it fire at this stage of testing. For consistency and sanity. I don't know how much work or the worth of it to prevent this auto-refresh from operations which may not need it such as failed ones. 

As it is, I'm glad flood control is disabled for this closed testing period or I'd be continually throttled out while re-trying to knock down this INVALID GREMLIN! Hopefully a resolution, or at least a reduction, can be made to this issue in the future.

**Comment by PlagueHush** [comment]
Started having this issue for the first time after updating from 0.62.10 to 0.62.17.
- Win 8.1
- D:\Nexus Mod Manager\NexusClient.exe
- 64-bit Process

Running in various compatibility modes and admin mode has not made any difference. Successful upload is still intermittent, which would seem to be a comms problem with the server.

It doesn't appear to make any difference if I'm uploading a new profile, or updating an existing one.

**Comment by KunoMochi** [comment]
PlagueHush, have you tried a fresh install of NMM by uninstalling NMM, remove all contents when asked, then reinstall it? You will have to reset your Mods Directory, Install Info, and Virtual Install paths when loading up NMM so remember to mark down where you have them currently.
